### PR TITLE
PP-3922 Added provider id and reference

### PR DESCRIPTION
**Pull request by georgeracu**
## WHAT

- Added `mandate_reference` from connector as `provider_id` to the services
that call Public API
- Added `service_reference` from connector as `reference` to the services
that call Public API
- Updated Swagger file

with @danailminchev

### CODE REVIEW

File: src/test/java/uk/gov/pay/api/utils/mocks/ConnectorDDMockClient.java [hunk 1]
```diff
@@ -35,63 +35,81 @@ public void respondWithChargeFound(
...
             String chargeTokenId) {
         String chargeResponseBody = buildPaymentRequestResponse(amount, chargeId, state, returnUrl,
-                description, reference, email, paymentProvider, createdDate,
-                validGetLink(chargeLocation(gatewayAccountId, chargeId), "self"),
-                validGetLink(nextUrl(chargeId), "next_url"),
-                validPostLink(nextUrlPost(), "next_url_post", "application/x-www-form-urlencoded",
-                        new HashMap<String, String>() {{
-                            put("chargeTokenId", chargeTokenId);
-                        }}));
+                                                                description, reference, email, paymentProvider, createdDate,
```
Comment (simad) [inline src/test/java/uk/gov/pay/api/utils/mocks/ConnectorDDMockClient.java]: wtf

Comment (simad) [reply src/test/java/uk/gov/pay/api/utils/mocks/ConnectorDDMockClient.java]: @DanailMinchev how could you allow this

Comment (DanailMinchev) [reply src/test/java/uk/gov/pay/api/utils/mocks/ConnectorDDMockClient.java]: @simad sorry :D evil auto-formatting and changed settings in the last minute

File: src/main/java/uk/gov/pay/api/model/directdebit/agreement/GetAgreementResponse.java [hunk 2]
```diff
@@ -8,25 +8,31 @@
...
+    private String providerId;
+    private String reference;
     private String returnUrl;
     private AgreementStatus state;
     private AgreementLinks links;
 
     private GetAgreementResponse(String agreementId,
                                  AgreementType agreementType,
-                                 String returnUrl,
+                                 String providerId, String reference, String returnUrl,
```
[the remark this answers is not in the document]
Comment (georgeracu) [reply src/main/java/uk/gov/pay/api/model/directdebit/agreement/GetAgreementResponse.java]: The mighty gods of code formatting were asleep as we reset IntelliJ's code formatting after the white space disaster. We promise not to happen again 🙏

**Pull request merged**